portal-impl/src/com/liferay/portal/service/persistence/impl/RoleFinderImpl.java

```diff
@@ -878,6 +1037,26 @@ protected String getCountByR_U_SQL() {
 		return _countByR_U;
 	}
 
+	protected String getExcludedNames(String[] excludedNames) {
+		int size = excludedNames.length;
+
+		for (String excludedName : excludedNames) {
+			if (excludedName == null) {
```

Comment:
When will "excludeName" ever be null? Also, if you pass in an array to the count/find methods, a lot of this logic becomes simpler.